Return nice error html/json when sandbox isn't found

0div opened the pull request:
# Description
- [x] introduce templated responses in `client-proxy`, similarly to what's done in [packages/orchestrator/internal/proxy/proxy.go](https://github.com/e2b-dev/infra/blob/0ffb276f93ea28acb27eafb2245c9442f0d90d46/packages/orchestrator/internal/proxy/proxy.go#L24)
  - [x] HTML template 
  - [x] JSON template
- [x] determine what response to return based on user agent 

### HTML Page 
![Screenshot 2025-03-26 at 4 03 41 PM](https://github.com/user-attachments/assets/471ba40c-fbbc-45d5-bfc8-88d8efe83a7f)

### JSON Response 
```
$ curl -i https://8080-ig6cw2it0s04dni6zvlzd-cf3f5359.goulash.dev/
HTTP/2 502
[...]
{"error":"Sandbox not found","sandboxId":"ig6cw2it0s04dni6zvlzd"}
```
# Test on your cluster
1. make build-and-upload/client-proxy
2. make plan
3. make apply
4.  navigate to `https://3000-ig6cw2it0s04dni6zvlzd-cf3f5359.{E2B_DOMAIN}.dev/` for HTML page
5. `curl -i https://3000-ig6cw2it0s04dni6zvlzd-cf3f5359.{E2B_DOMAIN}.dev/` for JSON response

r33drichards commented:
```
➜  ttemplate curl https://8080-ig6cw2it0s04dni6zvlzd-cf3f5359.e2b-robert.dev                             ~/ttemplate
{"error":"Sandbox not found","sandboxId":"ig6cw2it0s04dni6zvlzd"}%                                                   
➜  ttemplate        
```

<img width="1543" alt="image" src="https://github.com/user-attachments/assets/16f533c9-8269-4c39-abc4-c68f01982a11" />

r33drichards reviewed (approved):
lgtm! tested on my cluster

mlejva reviewed (changes requested):
@0div unify the style of the error message to the error message we have for the closed port error (see the screenshot for reference):

1. Add period at the end of the sentence "The sandbox <id> was not found".
2. Title "Sandbox not found" should have first letters of words uppercase, the same way 
we have "Closed Port Error".
3. Use the contraction as in the closed port error - it shouldn't be "The sandbox <id> was not found", it should be "The sandbox <id> wasn't found".

<img width="981" alt="Screenshot 2025-03-27 at 9 10 47 AM" src="https://github.com/user-attachments/assets/7a4d5af6-33df-4ef1-860e-6fac15deaf31" />

---


I'm also thinking removing the whole "Connection refused" section as that's not much helpful information anyway.

@0div I assume the visual theming matches the closed port error for both light and dark version?

sitole commented on packages/client-proxy/main.go:
```diff
@@ -92,10 +112,30 @@ func proxyHandler(transport *http.Transport) func(w http.ResponseWriter, r *http
+						w.WriteHeader(http.StatusInternalServerError)
+						return
+					}
+					w.WriteHeader(http.StatusBadGateway)
```
We should return 404 here to correctly communicate state. Same with JSON one

0div commented:
> @0div unify the style of the error message to the error message we have for the closed port error (see the screenshot for reference):
> 
> 1. Add period at the end of the sentence "The sandbox  was not found".
> 2. Title "Sandbox not found" should have first letters of words uppercase, the same way
>    we have "Closed Port Error".
> 3. Use the contraction as in the closed port error - it shouldn't be "The sandbox  was not found", it should be "The sandbox  wasn't found".
> 
> <img alt="Screenshot 2025-03-27 at 9 10 47 AM" width="981" src="https://private-user-images.githubusercontent.com/5136688/427408128-7a4d5af6-33df-4ef1-860e-6fac15deaf31.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDMwOTU2OTYsIm5iZiI6MTc0MzA5NTM5NiwicGF0aCI6Ii81MTM2Njg4LzQyNzQwODEyOC03YTRkNWFmNi0zM2RmLTRlZjEtODYwZS02ZmFjMTVkZWFmMzEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDMyNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAzMjdUMTcwOTU2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OTc5ZjY4NjUxNDM5NDE5NzU5NmNhZGIyMDE2ZDk5OGEwMjFjOGE3YjMwM2I3NmIwNTdiYzkxYWQzY2ExYmYzOCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ._7XeCzwvfWpJDTIbVCTHo5UanQ6mRVNAXO-dRtAwlSk">
> I'm also thinking removing the whole "Connection refused" section as that's not much helpful information anyway.
> 
> @0div I assume the visual theming matches the closed port error for both light and dark version?

@mlejva addressed all your points in this update, including returning 404 status code (cc @sitole), and yes, i copied the base html from the closed port error page so styling should be consistent:
